Add changelog to v3 core spec and update frontmatter

jstriebel opened the pull request:
This PR
* adds a changelog for the v3.0 core spec regarding implementation relevant changes. This is supposed to help implementations that track this spec to update with the spec.
* updates the frontmatter across different pages, mostly emphasizing the current status via a warning and updating some links. I also included some feedback about the frontmatter from #149.
* removes the version pinning for sphinx, as the old sphinx version does not work with current (unrestricted) versions of it's dependencies. The output of the current version conforms to the old one.

joshmoore commented on docs/codecs.rst:
```diff
@@ -29,9 +29,10 @@ This document defines codecs for Zarr implementations.
-or obsoleted by other documents at any time. It is inappapropriate to
-cite this document as other than work in progress.
+.. warning::
+    This document is a **Work in Progress**. It may be updated, replaced
```
:+1: though this comment made me think of a watermark so I filed https://github.com/zarr-developers/zarr-specs/issues/170

joshmoore commented on docs/core/v3.0.rst:
```diff
@@ -1810,8 +1812,25 @@ References
 
-This section is a placeholder for keeping a log of the snapshots of
-this document that are tagged in GitHub and what changed between them.
+All notable possibly implementation-affecting changes to this specification
```
```suggestion
All notable and possibly implementation-affecting changes to this specification
```

joshmoore commented on docs/core/v3.0.rst:
```diff
@@ -1810,8 +1812,25 @@ References
+- Added array storage transformers which can be configured per array via the
+  storage_transformers name in the array metadata.
+  `PR #134 <https://github.com/zarr-developers/zarr-specs/pull/134>`_
+- The changelog is incomplete before 2022, please refer to the commits on
```
:+1: for adding this but also for working towards named milestones that we can use to refer to publicly. i.e. "Which draft of the v3 is implemented in zarr-python? in xtensor? etc."

joshmoore reviewed (approved):
Thanks for getting this started, @jstriebel. :+1:

MSanKeys963 commented on docs/extensions/storage-transformers/sharding/v1.0.rst:
```diff
@@ -35,6 +35,15 @@ Sharding co-locates multiple chunks within a storage object, bundling them in
+
+.. warning::
+    This document is a draft for review and subject to changes.
+    It will become final when the `Zarr Enhancement Proposal (ZEP) 2 <https://zarr.dev/zeps/draft/ZEP0001.html>`_
```
There's a typo in the hyperlink here. It should point to ZEP0002 at https://zarr.dev/zeps/draft/ZEP0002.html.

jstriebel replied:
Good catch, thanks :+1: If you want to, you can even propose specific changes in a comment using the suggestion button:
![Screenshot from 2022-11-22 10-07-25](https://user-images.githubusercontent.com/7216331/203272608-15270b0d-f3d0-453c-bce5-8bb38b968e44.png)

They appear like below, and I can simply accept them to this PR:

```suggestion
    It will become final when the `Zarr Enhancement Proposal (ZEP) 2 <https://zarr.dev/zeps/draft/ZEP0002.html>`_
```

MSanKeys963 commented:
Hi @jstriebel. Thanks for sending the PR. 
All looks good to me, except for the comment I left above. 🚀

jstriebel commented:
Also added a change for the "Editor's draft <date>" titles: They used to be subtitles, which prevented the TOC (on the right) from expanding, since all other sections were a subpart of that subtitle. Changed it to not be a subtitle anymore, which directly shows all the elements in the TOC now.
